
From: Zwane Mwaikambo <zwane@fsmlabs.com>

William spotted this stray bit, LOCK_SECTION isn't used anymore on x86_64. 
Andrew i've diffed against -mm because i'd like for the irq enable on
contention patch to be merged, i believe making spinlock functions out of
line was a prerequisite Andi wanted before agreeing to the irq enable code.

Signed-off-by: Zwane Mwaikambo <zwane@fsmlabs.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/asm-x86_64/spinlock.h |   10 ++++------
 1 files changed, 4 insertions(+), 6 deletions(-)

diff -puN include/asm-x86_64/spinlock.h~remove-lock_section-from-x86_64-spin_lock-asm include/asm-x86_64/spinlock.h
--- 25/include/asm-x86_64/spinlock.h~remove-lock_section-from-x86_64-spin_lock-asm	Wed Sep 15 14:42:52 2004
+++ 25-akpm/include/asm-x86_64/spinlock.h	Wed Sep 15 14:42:52 2004
@@ -48,20 +48,18 @@ typedef struct {
 #define spin_lock_string \
 	"\n1:\t" \
 	"lock ; decb %0\n\t" \
-	"js 2f\n" \
-	LOCK_SECTION_START("") \
+	"jns 3f\n" \
 	"2:\t" \
 	"rep;nop\n\t" \
 	"cmpb $0,%0\n\t" \
 	"jle 2b\n\t" \
 	"jmp 1b\n" \
-	LOCK_SECTION_END
+	"3:\n\t"
 
 #define spin_lock_string_flags \
 	"\n1:\t" \
 	"lock ; decb %0\n\t" \
-	"js 2f\n\t" \
-	LOCK_SECTION_START("") \
+	"jns 4f\n\t" \
 	"2:\t" \
 	"test $0x200, %1\n\t" \
 	"jz 3f\n\t" \
@@ -72,7 +70,7 @@ typedef struct {
 	"jle 3b\n\t" \
 	"cli\n\t" \
 	"jmp 1b\n" \
-	LOCK_SECTION_END
+	"4:\n\t"
 
 /*
  * This works. Despite all the confusion.
_
